pkg/lib/tmp/copy.go

```diff
@@ -0,0 +1,49 @@
+package tmp
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+func CopyTmpDB(src string) (tmpName string, err error) {
+	tmp, err := ioutil.TempFile("", "db-")
+	if err != nil {
+		return
+	}
+	tmpName = tmp.Name()
+	err = CopyRegularFile(src, tmp.Name())
+	if err != nil {
+		err = utilerrors.NewAggregate([]error{err, os.Remove(tmpName)})
+	}
+	return
+}
+
+func CopyRegularFile(srcName, dstName string) (err error) {
+	srcStat, err := os.Stat(srcName)
+	if err != nil {
+		return err
+	}
+	if !srcStat.Mode().IsRegular() {
+		return fmt.Errorf("%s is not a regular file", srcName)
+	}
+	src, err := os.Open(srcName)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = utilerrors.NewAggregate([]error{err, src.Close()})
```

Comment:
We'd probably prefer to ignore it if the only error is on src.Close, at least for this use case.